fetch_cloud.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+)
+
+func FetchCloudData(lat, lon, apiKey string) (*http.Response, error) {
+	url := fmt.Sprintf("http://my.meteoblue.com/packages/clouds-1h_clouds-day?lat=%s&lon=%s&apikey=%s", lat, lon, apiKey)
+	client := http.Client{Timeout: 10 * time.Second}
+	resp, err := client.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to fetch data: status code %d", resp.StatusCode)
```

Comment:
Also, just because the status code isn't OK, u shouldn't send nil in the return value for *http.Response, as the resp is itself not nil, and doing so, u won't close the resp as you return prematurely (line 29 handlerGetCloud)

Reply:
yea my bad not closing the body

for the status code part i should restrict it to 200 to 300.

so i will change these